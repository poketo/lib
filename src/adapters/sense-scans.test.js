import poketo from '../index';
import site from './sense-scans';

describe('SenseScans', () => {
  const server = new AdapterVcrServer(site);

  beforeAll(async () => {
    await server.listenAndMock(57168);
  });

  afterAll(() => {
    server.close();
  });

  describe('getSeries', () => {
    it('returns a metadata object', async () => {
      const { chapters, ...metadata } = await poketo.getSeries(
        'sense-scans:kingdom',
      );

      expect(metadata).toMatchSnapshot();

      const chapterNumbersToTest = ['566', '568'];
      const chaptersToTest = chapters.filter(chapter =>
        chapterNumbersToTest.includes(chapter.chapterNumber),
      );
      expect(chaptersToTest).toMatchSnapshot();
    });
  });
});
