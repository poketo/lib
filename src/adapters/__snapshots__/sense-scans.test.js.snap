// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SenseScansAdapter getSeries returns a metadata object 1`] = `
Object {
  "chapters": ArrayContaining [
    Object {
      "createdAt": 1508059938,
      "number": "426.5",
      "slug": "en/39/426/5",
      "url": "https://sensescans.com/reader/read/kingdom/en/39/426/5/page/1",
    },
    Object {
      "createdAt": 1520340128,
      "number": "550",
      "slug": "en/51/550",
      "url": "https://sensescans.com/reader/read/kingdom/en/51/550/page/1",
    },
  ],
  "slug": "kingdom",
  "title": "Kingdom",
  "url": "https://sensescans.com/reader/series/kingdom",
}
`;
